### DISCUSSION TITLE
OSError: [Errno 5] EIO

### DISCUSSION THREAD

**Discussion by jpgpngexetxt** [micropython/micropython#15772] [RP2040 / Pico] [answered]
help me!
I was doing what was written on this page.
↓
https://www.waveshare.com/wiki/Pico-10DOF-IMU

An error occurred when running the MicroPython item.
Traceback (most recent call last):
  File "<stdin>", line 395, in <module>
  File "<stdin>", line 119, in __init__
  File "<stdin>", line 187, in configAK8963
OSError: [Errno 5] EIO

ips22hbtr.py  was working fine but mpu9250.py is not working.

[python.zip](https://github.com/user-attachments/files/16844592/python.zip)

please help Ultra beginner

thank you

**Comment by scruss** [comment, marked as answer]
According to the specification page you linked to, the board has a ICM20948 IMU sensor, not a MPU9250. The library is failing where it's trying to initialize a AK8963 magnetometer, which you maybe don't have 

what does an i2c.scan() tell you? An MPU-9250 has a default I2C address of 104 (0x68), while the default I2C address for the ICM20948 is 105 (0x69).

**Reply by jpgpngexetxt** [reply]
I was mistaken about which example program to run, I was able to get the results by running icm20948.py.

This was extremely helpful ! Thank you very much !
